[Embeddable] Fix typing workaround in the apply filter action

dokmic opened the pull request:
## Summary

Resolves #116894.

elasticmachine commented:
Pinging @elastic/kibana-app-services (Team:AppServicesUx)

vadimkibana reviewed (commented):
Could you please take a look at these other places too:

<img width="403" alt="image" src="https://user-images.githubusercontent.com/82822460/182596055-a6f9b488-291d-4c3f-a4f0-271887be1444.png">

dokmic commented:
@vadimkibana I've removed the data plugin workarounds but not the row click trigger. The latter still has a circular dependency and should be fixed entirely differently. That trigger should be encapsulated in the Lens plugin. I have already addressed that in #130281, but it is not ready to be merged.

vadimkibana reviewed (approved):
I see some tests are failing, but code LGTM.

dokmic commented:
@vadimkibana I had to revert fixes in the `data` plugin as they were failing the build. I tried to work the problem with circular references around, and there is no way we can do that without significant refactoring.
I think the best option would be to fix that like other actions in #130281. Since those changes are related, I would fix them as part of #111426.
